programs/compressed-token/src/lib.rs

```diff
@@ -0,0 +1,239 @@
+    //         utxo_hash,
+    //     )?;
+
+    //     self.frozen_utxo_pda.utxo_hash = utxo_hash;
+
+    //     Ok(())
+    // }
+
+    // #[derive(Accounts)]
+    // #[instruction(utxo_hash: [u8; 32])]
+    // pub struct InstructionFreezeUtxo {
+    //     pub fee_payer: Signer<'info>,
+    //     pub freez_authority: Signer<'info>,
+    //     /// Check that freeze_authority_pda is freeze authority
+    //     #[account(constraint= mint.freez_authority == *freez_authority.key())]
+    //     pub mint: Mint<'info, Mint>,
+    //     /// Used to sign transactions
+    //     #[account(seeds = [signer.key().to_bytes()], bump,)]
+    //     pub authority_pda: AccountInfo<'info>,
+    //     pub system_program: Program<'info, System>,
+    //     /// CHECK this account
+    //     pub registered_verifier_pda: AccountInfo<'info>,
+    //     pub merkle_tree_set: MerkleTreeSet<'info>,
+    //     #[account(init, seeds = [b"frozen_utxo", utxo_hash.as_ref()], bump, payer = fee_payer, space = 32)]
+    //     pub frozen_utxo_pda: Account<'info, FrozenUtxo>,
+    // }
+
+    // #[account]
+    // pub struct FrozenUtxo{
+    //     utxo_hash: [u8;32],
+    //     leaf_index: u64,
+    // }
+
+    // pub fn thaw_utxo<'info>(
+    //     ctx: Context<'_, '_, '_, 'info, InstructionThawUtxo<'info>>,
+    // ) -> Result<()> {
+
+    //     insert_utxo_into_state_tree_again(
+    //         &ctx.accounts.registered_verifier_pda,
+    //         &ctx.accounts.merkle_tree_set,
+    //         utxo_hash,
+    //         leaf_index,
+    //     )?;
+    //     Ok(())
+    // }
+
+    // #[derive(Accounts)]
+    // #[instruction(utxo_hash: [u8; 32])]
+    // pub struct InstructionThawUtxo {
+    //     pub fee_payer: Signer<'info>,
+    //     pub freez_authority: Signer<'info>,
+    //     /// Check that freez_authority_pda is freeze authority
+    //     #[account(constraint= mint.freez_authority == *freez_authority.key())]
+    //     pub mint: Mint<'info, Mint>,
+    //     /// Used to sign transactions
+    //     #[account(seeds = [signer.key().to_bytes()], bump,)]
+    //     pub authority_pda: AccountInfo<'info>,
+    //     /// CHECK this account
+    //     pub registered_verifier_pda: AccountInfo<'info>,
+    //     pub merkle_tree_set: MerkleTreeSet<'info>,
+    //     #[account(seeds = [b"frozen_utxo", frozen_utxo_pda.utxo_hash.as_ref()], bump, close=fee_payer)]
+    //     pub frozen_utxo_pda: Account<'info, FrozenUtxo>,
+    // }
+}
+
+// verifier sdk improvements
+// - bundle into verify, get public inputs, de_compress
+#[error_code]
+pub enum ErrorCode {
+    #[msg("public keys and amounts must be of same length")]
+    PublicKeyAmountMissmatch,
+    #[msg("missing new authority pda")]
+    MissingNewAuthorityPda,
+}
+
+// // TODO: unify these structures one way would be to put them into the registry program and import that into verifier sdk
+// // so that all accounts we use for serializeation in ts come from the registry program
```

Comment:
```suggestion
// // TODO: unify these structures. One way would be to put them into the registry program and import that into verifier sdk
// // so that all accounts we use for serialization in ts come from the registry program
```